x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -168,20 +168,32 @@ export const getCspStatus = async ({
     installedPackagePoliciesVulnMgmt,
     installedPolicyTemplates,
   ] = await Promise.all([
-    checkIndexStatus(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS, logger),
-    checkIndexStatus(esClient, FINDINGS_INDEX_PATTERN, logger),
-    checkIndexStatus(esClient, BENCHMARK_SCORE_INDEX_DEFAULT_NS, logger),
-
-    checkIndexStatus(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS, logger, 'cspm'),
-    checkIndexStatus(esClient, FINDINGS_INDEX_PATTERN, logger, 'cspm'),
-    checkIndexStatus(esClient, BENCHMARK_SCORE_INDEX_DEFAULT_NS, logger, 'cspm'),
-
-    checkIndexStatus(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS, logger, 'kspm'),
-    checkIndexStatus(esClient, FINDINGS_INDEX_PATTERN, logger, 'kspm'),
-    checkIndexStatus(esClient, BENCHMARK_SCORE_INDEX_DEFAULT_NS, logger, 'kspm'),
-
-    checkIndexStatus(esClient, LATEST_VULNERABILITIES_INDEX_DEFAULT_NS, logger),
-    checkIndexStatus(esClient, VULNERABILITIES_INDEX_PATTERN, logger, VULN_MGMT_POLICY_TEMPLATE),
+    checkIndexStatus(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS, logger, 'all', 72),
```

Comment:
I'm not sure I understand where these magic values are coming from. `72, 24, 5, 96` What do they mean, what is the logic behind them? I'd expect to see them somewhere else in the code and be reused here as constants, but I can't find them anywhere. Can you elaborate on that?

Reply:
those refers to the retention period where
CSPM has 24 hours retention period, KSPM 5 hours, CNVM 72 Hours, etc

Reply:
instead of using the numbers, I believe we should use constants defined [here](https://github.com/elastic/kibana/blob/df9fd9d0304a6d7c44c6b8542fb9943e29a3d0d2/x-pack/plugins/cloud_security_posture/common/constants.ts#L48) and make use of the constants that are already used by the transform methods.

Then we can use something like this (already considering @maxcold suggestion for the options params):

```js
checkIndexStatus(esClient, FINDINGS_INDEX_PATTERN, logger, 
	{ postureType: 'all', retentionPeriod:  LATEST_FINDINGS_RETENTION_POLICY }
}),
```

In that case `retentionPeriod` need to changed to string and the query range can be replaced by something like this:

```js
range: {
    '@timestamp': {
	      gte: `now-${retentionPeriod}`,
	      lte: 'now',
    },
 }
```

Reply:
yep, exactly, I got the retention part of course, what I meant it wasn't clear where those values are coming from and what is the logic behind them. eg. why 72 and not 70 or any other number? Basically what @opauloh said